src/main/java/teammates/storage/api/FeedbackSessionsDb.java

```diff
@@ -342,6 +342,45 @@ public void addStudentRespondent(String email, FeedbackSessionAttributes feedbac
         addStudentRespondents(emails, feedbackSession);
     }
 
+    /**
+     * Soft-deletes a specific feedback session by its name and course id.
+     * @return Soft-deletion time of the feedback session.
+     */
+    public Instant softDeleteFeedbackSession(String feedbackSessionName, String courseId)
+            throws EntityDoesNotExistException {
+        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
+        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, feedbackSessionName);
+
+        FeedbackSession sessionEntity = getFeedbackSessionEntity(feedbackSessionName, courseId);
+
+        if (sessionEntity == null) {
+            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT);
+        }
+
+        sessionEntity.setDeletedTime(Instant.now());
+        saveEntity(sessionEntity);
+
+        return sessionEntity.getDeletedTime();
+    }
+
+    /**
+     * Restores a specific soft deleted feedback session to feedback sessions table.
```

Comment:
`feedback sessions table.` is something UI specific, we should not bring it here.

Reply:
I do not quite understand why it has to use `setDeletedAt()`  `resetDeletedAt()` in LogicTest.

```
        assertFalse(course1OfInstructor.isCourseDeleted());

        Instant deletedAt = coursesLogic.moveCourseToRecycleBin(course1OfInstructor.getId());
        course1OfInstructor.setDeletedAt(deletedAt);
``` 

What is this for `course1OfInstructor.setDeletedAt(deletedAt)` ?

Reply:
One line below (`verifyPresentInDatastore(course1OfInstructor);`) is to verify whether the entity   (`course1OfInstructor`) exist in the database. If we don't update attribute of `course1OfInstructor`, the verification will fail as `course1OfInstructor` has different attributes (i.e. deletedAt) from the one in the database.

BTW, I am asking whether `setDeletedAt()` not `setDeletedAt(Instant)` can be removed or not :P

Reply:
It is still being used in several of the test cases but not in the logic.  

Should I change `setDeletedAt()` to `attributes.deletedAt = null` ?
Is this a good practice?

Reply:
Hmmm. Like this one in DB test.

![image](https://user-images.githubusercontent.com/13977235/51961135-b14fc480-2496-11e9-9dde-c90b813ff8ca.png)

I do think you can replace it with the new method `softDeleteSession`.